Source/Core/Core/PowerPC/BreakPoints.cpp

```diff
@@ -10,12 +10,57 @@
 #include <string>
 #include <vector>
 
+#include <expr.h>
+
 #include "Common/CommonTypes.h"
 #include "Common/DebugInterface.h"
 #include "Common/Logging/Log.h"
 #include "Core/Core.h"
 #include "Core/PowerPC/JitInterface.h"
 #include "Core/PowerPC/MMU.h"
+#include "Core/PowerPC/PowerPC.h"
+
+static bool ParseGPR(const char* name, int* index)
+{
+  if (name[0] == 'r' && isdigit(name[1]) &&
+      (name[2] == '\0' || isdigit(name[2]) && (name[3] == '\0')))
+  {
+    int idx = name[1] - '0';
+    if (name[2] != '\0')
+    {
+      idx = idx * 10 + (name[2] - '0');
+    }
+    if (idx < 32)
+    {
+      *index = idx;
+      return true;
+    }
+  }
+
+  *index = 0;
+  return false;
+}
+
+static bool EvaluateExpression(const std::string& expression)
+{
+  static expr_var_list vars = {0};
+  expr* e = expr_create(expression.c_str(), expression.length(), &vars, nullptr);
```

Comment:
```cpp
struct Deleter {
  void operator()(expr* expression) const {
    expr_destroy(expression, nullptr);
  }
};
using ExprPointer = std::unique_ptr<expr, Deleter>;

ExprPointer e{expr_create(expression.c_str(), expression.length(), &vars, nullptr)};
```

allows wrapping the c-style destruction in a manner that prevents any leaks from occurring if any other exits to the function are added in the future (through refactoring or feature additions otherwise).

Reply:
I was going to look up unique_ptr deleters (again) but you spared me the trouble. Thanks!